system/lib/emmalloc.c

```diff
@@ -555,17 +562,29 @@ static void initialize_emmalloc_heap()
   MAIN_THREAD_ASYNC_EM_ASM(console.log('initialize_emmalloc_heap()'));
 #endif
 
+#ifndef NDEBUG
+  // Even though this is a singlethreaded portion and we don't need critical section locks
+  // here, assertion checks in claim_more_memory()->validate_memory_regions() call verify
+  // that we do have it, so in MEMVALIDATE enabled build modes, make sure we have the lock.
+  MALLOC_ACQUIRE();
+#endif
+
   // Start with a tiny dynamic region.
   claim_more_memory(3*sizeof(Region));
+
+#ifndef NDEBUG
```

Comment:
That is because `emmalloc_blank_slate_from_orbit()` is purely an internal test function that users aren't using, whereas `initialize_emmalloc_heap()` is linked to by all code, so it benefits from optimizing away the redundant locks.